sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Source.scala

```diff
@@ -29,8 +30,13 @@ trait Source  {
   /** Returns the schema of the data from this source */
   def schema: StructType
 
+  /** Returns the maximum available offset for this source. */
+  def getOffset: Option[Offset]
```

Comment:
`getOffset` sounds like an immutable method. 

But if I understand right, `Source` should move to the next offset in this method. (`FileStreamSource` changes its `maxBatchId`). Maybe call it `advanceToNextOffset`?